src/unique_arc.rs

```diff
@@ -183,6 +183,33 @@ impl<T> UniqueArc<[MaybeUninit<T>]> {
     }
 }
 
+impl<H, T> UniqueArc<HeaderSlice<H, [MaybeUninit<T>]>> {
+    /// Creates an Arc for a HeaderSlice using the given header struct and allocated space
+    /// for an unitialized slice of length `len`.
+    pub fn from_header_and_uninit_slice(header: H, len: usize) -> Self {
+        let inner = Arc::allocate_for_header_and_slice(len);
+
+        unsafe {
+            // Write the header.
+            ptr::write(&mut ((*inner.as_ptr()).data.header), header);
```

Comment:
nit: safety comments, please

Reply:
specifically saying that the rest can stay uninitialized

Reply:
actually, sorry, this is unsound: making an `&mut` reference to uninitialized memory is not sound even if you only write to it.. Use `ptr::write` with a raw pointer.

You can use `addr_of_mut!()` if you want to use field access expressions.